torch/lobpcg.py

```diff
@@ -0,0 +1,562 @@
+"""Locally Optimal Block Preconditioned Conjugate Gradient methods.
+"""
+# Author: Pearu Peterson
+# Created: November 2019
+
+__all__ = ['lobpcg']
+
+import torch
+from . import _linalg_utils as _utils
+
+
+def lobpcg(A, B=None, k=1, X=None, n=None, iK=None, niter=1000, tol=None,
+           largest=True, tracker=None, method=None, **params):
+    """Find the k smallest (or largest) eigenvalues and the corresponding
+    eigenvectors of a symmetric positive defined generalized
+    eigenvalue problem using matrix-free LOBPCG methods.
+
+    This function is a front-end to the following LOBPCG algorithms
+    selectable via `method` argument:
+
+      `method="basic"` - the LOBPCG method introduced by Andrew
```

Comment:
nit: is there a more precise name that could be given to this version of the algorithm?

Reply:
Other options would be: "original", "classical", "knyazev2001", "plain".
Out of these, I'd prefer "plain" or "knyazev2001". Although the latter might be inconvenient to type. What do you think?

Reply:
Keeping "basic". Feel free to reopen the conversation if there are better suggestions.